pkg/http/listener.go

```diff
@@ -107,6 +96,30 @@ func (listener *httpListener) start() {
 		}
 	}
 
+	// Guess TLS protocol returns true if HTTP1.1 or
+	// HTTP2 methods are not found in the input buffer.
+	//
+	// This function if it returns true doesn't mean
+	// buffer has indeed TLS data, which needs
+	// to be further validated by using tls handshake.
+	guessTLSProtocol := func(buf []byte) bool {
```

Comment:
yes guessTLS is basically guessed after you have validated that its not HTTP that's why its guess. Otherwise if i can definitely tell then i would name it as isTLS().